bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java

```diff
@@ -0,0 +1,249 @@
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.types.Command;
+import org.openwebnet4j.communication.OWNException;
+import org.openwebnet4j.message.Alarm;
+import org.openwebnet4j.message.Alarm.WhatAlarm;
+import org.openwebnet4j.message.BaseOpenMessage;
+import org.openwebnet4j.message.Where;
+import org.openwebnet4j.message.WhereAlarm;
+import org.openwebnet4j.message.Who;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenWebNetAlarmHandler} is responsible for handling
+ * commands/messages for Alarm system and zones. It extends the abstract
+ * {@link OpenWebNetThingHandler}.
+ *
+ * @author Massimo Valla - Initial contribution
+ */
+@NonNullByDefault
+public class OpenWebNetAlarmHandler extends OpenWebNetThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWebNetAlarmHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = OpenWebNetBindingConstants.ALARM_SUPPORTED_THING_TYPES;
+
+    private static long lastAllDevicesRefreshTS = 0; // ts when last all device refresh was sent for this handler
+
+    private static final String BATTERY_OK = "OK";
+    private static final String BATTERY_FAULT = "FAULT";
+    private static final String BATTERY_UNLOADED = "UNLOADED";
+
+    private static final String SILENT = "SILENT";
+    private static final String INTRUSION = "INTRUSION";
+    private static final String ANTI_PANIC = "ANTI_PANIC";
+    private static final String TAMPERING = "TAMPERING";
+
+    public OpenWebNetAlarmHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected void handleChannelCommand(ChannelUID channel, Command command) {
+        logger.warn("Alarm.handleChannelCommand() Read only channel, unsupported command {}", command);
+    }
+
+    @Override
+    protected void requestChannelState(ChannelUID channel) {
+        super.requestChannelState(channel);
+        Where w = deviceWhere;
+        ThingTypeUID thingType = thing.getThingTypeUID();
+        try {
+            if (THING_TYPE_BUS_ALARM_SYSTEM.equals(thingType)) {
+                send(Alarm.requestSystemStatus());
+                lastAllDevicesRefreshTS = System.currentTimeMillis();
+            } else {
+                if (w != null) {
+                    send(Alarm.requestZoneStatus(w.value()));
+                } else {
+                    logger.debug("null where while requesting state for channel {}", channel);
+                }
+            }
+        } catch (OWNException e) {
+            logger.debug("Exception while requesting state for channel {}: {} ", channel, e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    protected long getRefreshAllLastTS() {
+        return lastAllDevicesRefreshTS;
+    };
+
+    @Override
+    protected void refreshDevice(boolean refreshAll) {
+        if (refreshAll) {
```

Comment:
No need to check that thing type is THING_TYPE_BUS_ALARM_SYSTEM there ?

Reply:
in fact no: if refreshAll is true, the state for all alarm devices (both central unit and zones) will be performed by asking a complete alarm system status